cranelift/interpreter/src/value.rs

```diff
@@ -484,9 +489,15 @@ impl DataValueExt for DataValue {
 
     fn is_zero(&self) -> ValueResult<bool> {
         match self {
+            DataValue::I8(f) => Ok(*f == 0),
+            DataValue::I16(f) => Ok(*f == 0),
+            DataValue::I32(f) => Ok(*f == 0),
+            DataValue::I64(f) => Ok(*f == 0),
+            DataValue::I128(f) => Ok(*f == 0),
             DataValue::F32(f) => Ok(f.is_zero()),
             DataValue::F64(f) => Ok(f.is_zero()),
-            _ => Err(ValueError::InvalidType(ValueTypeClass::Float, self.ty())),
+            DataValue::V64(f) => Ok(f.iter().any(|e| *e == 0)),
+            DataValue::V128(f) => Ok(f.iter().any(|e| *e == 0)),
```

Comment:
I'm not very comfortable with this definition of `is_zero` for SIMD types. Using `any` means that we can have non zero values in a lane and this can still return true. And it would still be wrong for Float SIMD types where we can have a zero mantissa or exponent byte while the overall value is still non zero.

I think the best option here would be to return `InvalidType` for `V64` and `V128` since we don't really know which SIMD type this `DataValue` represents.

Reply:
Yes, you are right. I've used ValueTypeClass::Float for both Vector-Types Error-reply. This was the default anyway before adding all the branches.

Fixed with: https://github.com/ms140569/wasmtime/commit/0bd255231021a49eb944b27aaa66049056b5f39c